### PR TITLE
fix: lerna init should always use current working directory

### PR DESCRIPTION
**Pull request by noherczeg**

### CODE REVIEW

**Comment by noherczeg** [comment]
In response to #645

**Comment by noherczeg** [comment]
I'm not sure about the current implementation though. What I was thinking about was leaving GitUtilities as it was, but in InitCommand:
```
constructor(input, flags) {
    super(input, flags);
    this.repository.rootPath = process.cwd();
}
```
My problem was though that the parent's constructor already did tons of stuff with the original value at that point. So some insight for this would be very much appreciated!

File: src/GitUtilities.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ export default class GitUtilities {
 
   @logger.logifySync()
   static getTopLevelDirectory() {
-    return ChildProcessUtilities.execSync("git rev-parse --show-toplevel");
+    return process.cwd();
```
Comment (evocateur) [inline src/GitUtilities.js]: I think this is the wrong place to change the evaluation of a given Command's `rootPath` property. In fact, I don't think `Repository` should _ever_ be calling `git init`:

```diff
diff --git a/src/Repository.js b/src/Repository.js
index c9870f1..9ab637c 100644
--- a/src/Repository.js
+++ b/src/Repository.js
@@ -1,24 +1,20 @@
-import GitUtilities from "./GitUtilities";
+import path from "path";
+import findUp from "find-up";
 import FileSystemUtilities from "./FileSystemUtilities";
 import PackageUtilities from "./PackageUtilities";
 import Package from "./Package";
 import NpmUtilities from "./NpmUtilities";
-import path from "path";
-import logger from "./logger";
 
 const DEFAULT_PACKAGE_GLOB = "packages/*";
 
 export default class Repository {
   constructor() {
-    if (!GitUtilities.isInitialized()) {
-      logger.info("Initializing Git repository.");
-      GitUtilities.init();
-    }
+    // findUp returns null when not found, and path.resolve starts from process.cwd()
+    const lernaJsonLocation = findUp.sync("lerna.json") || path.resolve("lerna.json");
 
-    this.rootPath = path.resolve(GitUtilities.getTopLevelDirectory());
-    this.lernaJsonLocation = path.join(this.rootPath, "lerna.json");
+    this.rootPath = path.dirname(lernaJsonLocation);
+    this.lernaJsonLocation = lernaJsonLocation;
     this.packageJsonLocation = path.join(this.rootPath, "package.json");
-    this.packagesLocation = path.join(this.rootPath, "packages"); // TODO: Kill this.
 
     // Legacy
     this.versionLocation = path.join(this.rootPath, "VERSION");
```

Instead, `lerna init` should have that responsibility, as befits an "init" command. Normal commands don't need that baggage/overhead:

```diff
diff --git a/src/commands/InitCommand.js b/src/commands/InitCommand.js
index b41c462..f6c04d0 100644
--- a/src/commands/InitCommand.js
+++ b/src/commands/InitCommand.js
@@ -1,3 +1,4 @@
+import GitUtilities from "./GitUtilities";
 import FileSystemUtilities from "../FileSystemUtilities";
 import Command from "../Command";
 import objectAssignSorted from "object-assign-sorted";
@@ -8,7 +9,11 @@ export default class InitCommand extends Command {
   runPreparations() {}
 
   initialize(callback) {
-    // Nothing to do...
+    if (!GitUtilities.isInitialized()) {
+      this.logger.info("Initializing Git repository.");
+      GitUtilities.init();
+    }
+
     callback(null, true);
   }
```

Comment (noherczeg) [inline src/GitUtilities.js]: @evocateur Sure, np! :) Glad I could help!

Comment (noherczeg) [inline src/GitUtilities.js]: @evocateur You can close this if it is no longer necessary!

**Pull request closed without merging**